src/main/java/org/lscode/DupLocator/FileStorage.java

```diff
@@ -0,0 +1,94 @@
+/**
+ *  file:    FileStorage.java
+ *  desc:    concrete class for storing FileArray objects segregated according to a chosen criterion
+ *  author:  lscode
+ *  license: GNU General Public License v3.0
+ */
+
+package org.lscode.DupLocator;
+
+import java.util.*;
+
+public class FileStorage<T> implements Iterable<FileArray>{
+
+    protected Map<T, FileArray> fileMap = new HashMap<>();
+
+//    public FileStorage(){}
+
+//    public FileStorage(FileStorage<T> source){
+//    }
+
+    public Iterator<FileArray> iterator() {
+        return fileMap.values().iterator();
+    }
+
+    public void put(T criterion, FileData fileData){
+        FileArray fileList;
+
+        if (fileMap.containsKey(criterion)){
+            fileList = fileMap.get(criterion);
+        }
+        else{
+            fileList = new FileArray();
+            fileMap.put(criterion, fileList);
+        }
+        fileList.add(fileData);
+    }
+
+    public void putArray(T criterion, FileArray fArray){
+        if (fileMap.containsKey(criterion)){
```

Comment:
nie rozumiem tej metody, po co rozbiles to na dwa ify? Wydaje mi sie ze implementacja Map ma odpowiednia metode do tego (dodaj nowy jezeli nie ma w przeciwnym wypadku update)